cpp/src/Ice/Instance.cpp

```diff
@@ -1112,9 +1125,9 @@ IceInternal::Instance::initialize(const Ice::CommunicatorPtr& communicator)
 
         {
             int32_t num = _initData.properties->getIcePropertyAsInt("Ice.ClassGraphDepthMax");
-            if (num < 1 || static_cast<size_t>(num) > static_cast<size_t>(0x7fffffff))
+            if (num < 1)
             {
-                const_cast<size_t&>(_classGraphDepthMax) = static_cast<size_t>(0x7fffffff);
+                const_cast<size_t&>(_classGraphDepthMax) = static_cast<size_t>(size_max);
```

Comment:
This is unrelated to messageSizeMaxUpperLimit. Just use `std::numeric_limits<int32_t>::max()`.